aclocal.m4

```diff
@@ -643,3 +643,23 @@ AC_DEFUN([OCAML_CHECK_WINDOWS_TRIPLET], [
     [*-pc-windows*],
       [AC_MSG_ERROR([unknown MSVC variant])])
 ])
+
+# It's difficult to use AC_PROG_CXX or AX_CXX_COMPILE_STDCXX conditionally.
+# This macro is only used for ocamltest to call the C++11 compiler if the
+# default C compiler also can build C++.
+AC_DEFUN([OCAML_CXX_COMPILE_STDCXX_11], [
+  AC_CACHE_CHECK([for a C++11 compiler],
+    [ocaml_cv_prog_cxx], [
+    AS_IF([test "$ccomp_type" = "cc"], [
+      saved_CC="$CC"
+      CC="$CC -xc++"
+      AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#if !defined(__cplusplus) || __cplusplus < 201103L
+#error "No C++11 support"
+#endif
+      ]])],
+      [ocaml_cv_prog_cxx="$CC"],
+      [ocaml_cv_prog_cxx=""])
+      CC="$saved_CC"],
+    [ocaml_cv_prog_cxx="$CC"])])
```

Comment:
I found a bit subtle that this is "cl always C++" - perhaps a comment that MSVC always supports C++ here?